src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -66,6 +66,18 @@ public override HttpResponseMessage CreateCheckStatusResponse(HttpRequestMessage
                 functionType: FunctionType.Orchestrator,
                 isReplay: false);
 
+            if (waitUntilOrchestrationStarts)
+            {
+                DurableOrchestrationStatus status = await this.GetStatusAsync(instance.InstanceId);
+                int counter = 0;
```

Comment:
The problem with using a loop and a counter is that the timeout cannot be measured accurately. For example, if `GetStatusAsync` is slow for any reason, callers might end up waiting much longer than 10 seconds. Instead, I suggest using a `Stopwatch`. For example:

```csharp
var stopwatch = Stopwatch.StartNew();
while ((status == null || status.RuntimeStatus == OrchestrationRuntimeStatus.Pending) && stopwatch.Elapsed < TimeSpan.FromSeconds(10))
{
    await Task.Delay(200);
    status = await this.GetStatusAsync(instanceId);
}
```

Reply:
Thank you for the code snippet! Great improvement!